### ISSUE TITLE
Looking for Help to Understand the Decoder Code

### ISSUE THREAD

**Issue by ShayanShamsi** [xi-jia/Fourier-Net#6]
Hi. Thank you for the amazing work you put out.

I am looking for some help to understand the decoder code. Looking at Models.py 3D FourierNet, I saw that the decoder is represented in the forward pass as:
<pre><code class="language-python">
        r_d0 = torch.cat((self.r_up1(e4), e3), 1)

        r_d0 = self.r_dc1(r_d0)
        r_d0 = self.r_dc2(r_d0)

        r_d1 = torch.cat((self.r_up2(r_d0), e2), 1)

        r_d1 = self.r_dc3(r_d1)
        r_d1 = self.r_dc4(r_d1)
        f_r = self.rr_dc9(r_d1)
        return f_r[:,0:1,:,:,:], f_r[:,1:2,:,:,:], f_r[:,2:3,:,:,:]
</code></pre>

Can you please help me understand how this represents a model-driven decoder consisting of a zero-padding layer, an iDFT layer, and an optional squaring and scaling module as explained in the paper?

**Comment by xi-jia** [comment]
`how this represents a model-driven decoder consisting of a zero-padding layer, an iDFT layer, and an optional squaring and scaling module as explained in the paper?`

Please see [train.py](https://github.com/xi-jia/Fourier-Net/blob/b329d18e101b07a75fad2986d4686ebc1992b21c/3D%20Fourier-Net/train.py#L164) and [infer_bilinear.py](https://github.com/xi-jia/Fourier-Net/blob/b329d18e101b07a75fad2986d4686ebc1992b21c/3D%20Fourier-Net/infer_bilinear.py#L120) for the details.

[This link](https://github.com/xi-jia/Fourier-Net/blob/b329d18e101b07a75fad2986d4686ebc1992b21c/3D%20Fourier-Net%2B/Models.py#L49) might be helpful if you prefer an embedded module within the network architecture.

**Comment by ShayanShamsi** [comment]
I overlooked that. Thanks for providing the clarification!
